File: Backend/airport-backend/src/main/java/vn/iuhcm/airport/passenger/Passenger.java
package vn.iuhcm.airport.passenger;

public class Passenger {

    public String passengerID;   // PA00001
    public String passportID;
    public String firstName;
    public String middleName;
    public String lastName;
    public String nation;
    public String email;

    // Constructor rỗng (Jackson cần)
    public Passenger() {}

    // Constructor đầy đủ
    public Passenger(String passengerID, String passportID, String firstName,
                     String middleName, String lastName, String nation, String email) {
        this.passengerID = passengerID;
        this.passportID = passportID;
        this.firstName = firstName;
        this.middleName = middleName;
        this.lastName = lastName;
        this.nation = nation;
        this.email = email;
    }

    // Optional: Getter cho Fullname
    public String getFullname() {
        if (middleName == null || middleName.isBlank()) {
            return firstName + " " + lastName;
        }
        return firstName + " " + middleName + " " + lastName;
    }
}

File: Frontend/airport-frontend/app/types/Passenger.ts
// File: app/types/Passenger.ts

export interface Passenger {
  /** Primary Key: PA00001 */
  passengerID: string;

  /** Unique passport identifier */
  passportID: string;

  /** First name */
  firstName: string;

  /** Middle name (optional) */
  middleName: string | null;

  /** Last name */
  lastName: string;

  /** Nationality */
  nation: string | null;

  /** Email address */
  email: string | null;
}

